### ISSUE TITLE
archival node unable to sync

### ISSUE THREAD

**Issue by bowenwang1996** [near/nearcore#6266]
I observed that when running an archival node with this branch [https://github.com/near/nearcore/tree/1.24.0-fix](https://github.com/near/nearcore/tree/1.24.0-fix) something very strange happened. The node stopped syncing at height 59025077 which is the last block of an epoch. Upon investigation I discovered that the very next block is orphaned because the node thinks that it does not have state to apply the block (the logic defined [here](https://github.com/near/nearcore/blob/e2603ee1435cf1614537b30fabb61bdfaac2d971/chain/chain/src/chain.rs#L4102)). This doesn't make any sense: the node has never been a validator node and it does not need to run the catchup logic anyways. It is not immediately clear to me what exactly is wrong here, but potentially there is some bug in the catchup logic and blocks are not removed properly. More specifically, we have this questionable logic that always considers the first block of the epoch "not caught up" and if it is not removed properly it could lead to the aforementioned behavior.

**Comment by bowenwang1996** [comment]
We saw this on the mainnet achival canary node again. cc @mina86 @mzhangmzz

**Comment by mzhangmzz** [comment]
I did some digging, unfortunately I accidentally overwrote the log that I saved and now I don't have log any more. I think the issue is that for some reason, state sync in the catchup phase is taking a long time (more than an epoch) even though there is no state to download, which is weird. It's hard for me to tell right now whether https://github.com/near/nearcore/blob/master/chain/client/src/client.rs#L1701 is taking a long time or https://github.com/near/nearcore/blob/master/chain/client/src/client.rs#L1717 is taking a long time. I'll submit a PR that adds more debug prints to print more information.
